src/test/java/com/puppycrawl/tools/checkstyle/api/FileSetCheckTest.java

```diff
@@ -40,8 +39,8 @@ protected String getPackageLocation() {
     public void testTranslation() throws Exception {
         final Configuration checkConfig =
             createModuleConfig(TestFileSetCheck.class);
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verify(checkConfig, getPath("InputFileSetIllegalTokens.java"), expected);
+        final String path = getPath("InputFileSetIllegalTokens.java");
+        execute(checkConfig, path);
 
```

Comment:
hi @romani , i am getting one exception when i go with **verifyWithInlineConfigParser(getPath("InputFileSetIllegalTokens.java"), expected);** although there is no violation set in InputFileSetIllegalTokens.java
this is expected **final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;**

exception: com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - TreeWalker is not allowed as a parent of com.puppycrawl.tools.checkstyle.api.FileSetCheckTest$TestFileSetCheck Please review 'Parent Module' section for this Check in web documentation if Check is standard.

![Screenshot 2023-12-03 162721](https://github.com/checkstyle/checkstyle/assets/76208964/53191dc9-d7a3-4acf-8213-4a17ccf2e932)
![Screenshot 2023-12-03 162951](https://github.com/checkstyle/checkstyle/assets/76208964/fbe4359f-1020-46bc-9ee7-df233b92250b)

Reply:
please debug creation of config before exeuction, there should be some hardcoded map of Checks that must be under Checker not a Treewalker.
as alternative you can try to use xml form of config in comment as we use in Examples.

Reply:
hi @romani, i debuged the issue also i debugged the above commit code, and i see there also it is going under setupchild method of treewalker in both code( mine and above commit code) i am not able to figuring out where is the issue, this is the generated config which i can see
![Screenshot 2023-12-05 085324](https://github.com/checkstyle/checkstyle/assets/76208964/0d2cf9cc-f764-44e9-85bb-6e674544a232)

in this commit changes https://github.com/checkstyle/checkstyle/commit/7f50a149922ba81ae867f967e8f146b3fa42097a#diff-93172eb00d9c0952dd2a9582f6048ef1e05e1f433b66a2ba0924b225f3b679b6R2 also when i debuged it the same config i am able to see 

And my code is breaking inside Treewalker, those two checks are failing because of that above exception is getting, 
but in that above commit, those checks are passing as true.
![Screenshot 2023-12-05 085930](https://github.com/checkstyle/checkstyle/assets/76208964/af52c63a-7f18-4c03-857a-df9812b0561d)

i also used both config type, inline and xml also
these are the config i am using
inline: 
**/*
com.puppycrawl.tools.checkstyle.api.FileSetCheckTest$TestFileSetCheck

*/**

xml type: 
![image](https://github.com/checkstyle/checkstyle/assets/76208964/bf305d35-ef9b-4f7f-895a-caf842c11f30)


this is the checker class
in this setupChild method exception is catching on line no 484 after throwing from Treewalker class setup child method line no 140 above is the ss i provided

Reply:
please push latest code from your local to github to let us see state of your code, we might hint on what to do on the go.

Reply:
sure

Reply:
pushed